x/evm/keeper/msg_server_user_smart_contract.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/palomachain/paloma/x/evm/types"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+func (k Keeper) UploadUserSmartContract(
+	ctx context.Context,
+	req *types.MsgUploadUserSmartContractRequest,
+) (*types.MsgUploadUserSmartContractResponse, error) {
+	contract := &types.UserSmartContract{
+		ValAddress:       req.ValAddress,
+		Title:            req.Title,
+		AbiJson:          req.AbiJson,
+		Bytecode:         req.Bytecode,
+		ConstructorInput: req.ConstructorInput,
+	}
```

Comment:
I suggest performing a bit of arbitrary input validation here. Take a look at what's happening with registering a new job. ABI is packed and input and bytecode are verified before accepting the job. I think it makes sense to do something similar here in order to avoid storing faulty contracts and wasting space.